generate_k_shot_data.py

```diff
@@ -161,28 +164,56 @@ def main_for_gao(args, tasks):
             if task in ["MNLI", "MRPC", "QNLI", "QQP", "RTE", "SNLI", "SST-2", "STS-B", "WNLI", "CoLA"]:
                 # GLUE style
                 # Use the original development set as the test set (the original test sets are not publicly available)
-                for split, lines in dataset.items():
-                    if split.startswith("train"):
-                        continue
-                    lines = dataset[split]
-                    split = split.replace('dev', 'test')
-                    with open(os.path.join(setting_dir, f"{split}.tsv"), "w") as f:
-                        for line in lines:
-                            f.write(line)
+                if not args.test_limit:
+                    for split, lines in dataset.items():
+                        if split.startswith("train"):
+                            continue
+                        lines = dataset[split]
+                        split = split.replace('dev', 'test')
+                        with open(os.path.join(setting_dir, f"{split}.tsv"), "w") as f:
+                            for line in lines:
+                                f.write(line)
+                else:
+                    for split, lines in dataset.items():
+                        if split.startswith("train"):
+                            continue
+                        if args.test_limit < len(lines):
+                            lines = random.sample(dataset[split], args.test_limit)
+                            lines = dataset[split][:args.test_limit]
+                        else:
+                            lines = dataset[split]
+                        split = split.replace('dev', 'test')
+                        with open(os.path.join(setting_dir, f"{split}.tsv"), "w") as f:
+                            for line in lines:
+                                f.write(line)
 
 
             else:
                 # Other datasets
                 # Use the original test sets
-                dataset['test'].to_csv(os.path.join(setting_dir, 'test.csv'), header=False, index=False)
+                if not args.test_limit or args.test_limit >= len(dataset['test']):
+                    dataset['test'].to_csv(os.path.join(setting_dir, 'test.csv'), header=False, index=False)
+                else:
+                    dataset['test'].sample(args.test_limit).to_csv(os.path.join(setting_dir, 'test.csv'), header=False, index=False)
```

Comment:
It is probably better to refactor to avoid duplicate code `.to_csv(os.path.join(setting_dir, 'test.csv'), header=False, index=False)`